src/hssm/param/regression_param.py

```diff
@@ -0,0 +1,402 @@
+    ) -> None:
+        """Fill in the default values for the parameter.
+
+        Parameters
+        ----------
+        formula
+            The formula for the parameter.
+        bounds
+            The bounds for the parameter.
+        prior_settings
+            The prior settings for the parameter.
+        link_settings
+            The link settings for the parameter.
+        """
+        if prior is not None:
+            raise ValueError(
+                f"{self.name} is a regression parameter. "
+                + "It should not have a default prior."
+            )
+        super().fill_defaults(prior=None, bounds=bounds)
+        if self.formula is None:
+            if "formula" not in kwargs:
+                raise ValueError(f"Formula not specified for parameter {self.name}.")
+            self.formula = kwargs["formula"]
+        if kwargs.get("link_settings") == "log_logit":
+            self.set_loglogit_link()
+
+    def validate(self) -> None:
+        """Validate the parameter."""
+        if self.formula is None:
+            raise ValueError(f"Formula not specified for parameter {self.name}.")
+        self.reformat_formula()
+        if isinstance(self.prior, bmb.Prior):
+            raise ValueError(
+                "Please specify priors for each individual parameter in the "
+                f"regression for {self.name}."
+            )
+        if self.link is None:
+            self.link = "identity"
+        else:
+            if self.link == "log_logit":
+                self.link = Link("gen_logit", bounds=self.bounds)
+
+    def process_prior(self) -> None:
+        """Process the prior specification."""
+        self.validate()
+        if self.prior is None:
+            return
+        self.prior = cast(dict[str, Any], self.prior)
+        self.prior = _make_prior_dict(self.prior)
+
+    def set_loglogit_link(self):
+        """Override the default link function.
+
+        This is most likely because both default prior and default bounds are supplied.
+        """
+        if self.link is not None:
+            return
+
+        if self.bounds is None:
+            raise ValueError(
+                "Cannot override the default link function. "
+                f"Bounds are not specified for parameter {self.name}."
+            )
+
+        lower, upper = self.bounds
+
+        if np.isneginf(lower) and np.isposinf(upper):
+            self.link = "identity"
+        elif lower == 0.0 and np.isposinf(upper):
+            self.link = "log"
+        elif not np.isneginf(lower) and not np.isposinf(upper):
+            self.link = Link("gen_logit", bounds=self.bounds)
+        else:
+            _logger.warning(
+                "The bounds for parameter %s (%f, %f) seem strange. Nothing is done to"
+                + " the link function. Please check if they are correct.",
+                self.name,
+                lower,
+                upper,
+            )
+
+    def make_safe_priors(
```

Comment:
This method seems to be doing a lot of things. You might want to consider splitting it into a couple of methods, one for each of the main blocks of code, or perhaps add some high-level comments describing the what/why for each?

Reply:
Added some high level comments